File: programme/array-elements-slice/array-elements-slice.php
<?php
// Write a programme to generate new array with sliced elements of old array

// define array
$array_list = ['abcd', 'efgh', 'ijklmn', 'opq', 'rstuvw', 'xyz'];
print 'Array : ' . json_encode($array_list);

# function arguments (arrayList) (start slice from -number) (end slice with -number)
function arrayLetterRange($a_list, $start = 1, $end = 2)
{
    // store slice value in result
    $result = [];
    for ($i = 0; $i < sizeof($a_list); $i++) {
        $sliced_val = substr($a_list[$i], $start - 1, $end);

        // check slice value if not empty
        if ($sliced_val != '') {
            array_push($result, $sliced_val);
        }
    }
    return $result;
}

// define range
$start = 2;
$end = 3;
print "\nStart : $start";
print "\nEnd   : $end";

// call the function
$array_result = arrayLetterRange($array_list, $start, $end);
print "\n\nSliced Array : " . json_encode($array_result);
print "\n";

?>

File: programme/array-elements-slice/ArrayElementsSlice.java
// Write a programme to generate new array with sliced elements of old array

import java.util.Arrays;

public class ArrayElementsSlice {

  public static void main(String[] args) {
    String[] array = new String[] {
      "abcd",
      "efgh",
      "ijklmn",
      "opq",
      "rstuvw",
      "xyz",
    };
    int start = 2;
    int end = 3;
    System.out.println("Start index : " + start);
    System.out.println("End   index : " + end);
    System.out.println(
      "\nOutput : " + Arrays.toString(arrayElementsSlice(array, start, end))
    );
  }

  private static String[] arrayElementsSlice(
    String[] array,
    int start,
    int end
  ) {
    for (int i = 0; i < array.length; i++) {
      array[i] = array[i].substring(start - 1, end);
    }
    return array;
  }
}
